lib/fs/filesystem.go

```diff
@@ -178,9 +179,9 @@ var IsPermission = os.IsPermission
 // IsPathSeparator is the equivalent of os.IsPathSeparator
 var IsPathSeparator = os.IsPathSeparator
 
-type Option struct {
-	apply func(Filesystem)
-	id    string
+type Option interface {
+	ID() string
```

Comment:
```suggestion
	String() string
```

Might as well be a stringer? Feels more like a string representation of an option, especially if there's a value, than a unique ID of some sort.

Reply:
It has to be IDish, as in every option with the same `ID` should have the same effect. That's required for the case-fs caching. I imagined that if an option has a parameter, it needs to be a query-like string (`name=param` if only a single parameter is allowed, or `name?param1=value1&param2=value2`). Implementing `Stringer` is anyways always a good thing, will do, I'd still like to keep `ID` to signify in the method name that it is more than just an "informal" string representation. I added a comment explaining the purpose of `ID` to the `Option` interface.  
Does that make sense? Or maybe be more specific and have both a `Name() string` and `Params() uri.Values` method on `Options` - feels a bit over-engineered though.

Reply:
Right, that's where I was thinking that intuitively to me for some hypothetical optional like `lstatCacheTime=30s` the "ID" would be `statCacheTime` and the string representation would be `lstatCacheTime=30s` and it's the latter we would use when caching because two options with the same ID but with different values are different. But we don't have that today and the distinction doesn't exist yet so it's all hypothetical and based on feeling. :)

Reply:
`String` it is - and if it ever gets non-hypothetical, we can reopen the discussion (or not :) ).